src/java.net.http/share/classes/jdk/internal/net/http/Stream.java

```diff
@@ -652,10 +649,16 @@ void incoming_pushPromise(HttpRequestImpl pushRequest,
         // Filter context restricted from userHeaders
         userh = HttpHeaders.of(userh.map(), Utils.CONTEXT_RESTRICTED(client()));
 
+        // Don't override Cookie values that have been set by the CookieHandler.
         final HttpHeaders uh = userh;
+        BiPredicate<String, String> overrides =
+                (k, v) -> COOKIE_HEADER.equalsIgnoreCase(k)
+                          || uh.firstValue(k).isEmpty();
 
         // Filter any headers from systemHeaders that are set in userHeaders
-        sysh = HttpHeaders.of(sysh.map(), (k,v) -> uh.firstValue(k).isEmpty());
+        //   except for "Cookies" - user cookies will be appended to system
```

Comment:
A minor comment issue - "Cookie" instead of "Cookies", perhaps?

Reply:
Changed to "Cookie:"